src/System.Runtime.Extensions/tests/System/Environment.ProcessorCount.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Runtime.InteropServices;
+using Xunit;
+
+namespace System.Runtime.Extensions.Tests
+{
+    public class EnvironmentProcessorCount
+    {
+        [Fact]
+        public void ProcessorCountTest()
+        {
+            //arrange
+            int expected;
+
+            if(Interop.IsWindows)
+            {
+                SYSTEM_INFO sysInfo = new SYSTEM_INFO();
+                SYSTEM_INFO.GetSystemInfo(ref sysInfo);
+                expected = sysInfo.dwNumberOfProcessors;
+            }
+            else
+            {
+                int _SC_NPROCESSORS_ONLN = Interop.IsOSX ? 58 : 84;
+                expected = (int)sysconf(_SC_NPROCESSORS_ONLN);                
+            }
+
+            //act
+            int actual = Environment.ProcessorCount;
+
+            //assert
+            Assert.True(actual > 0);
+            Assert.Equal(expected, actual);
+        }
+
+        [DllImport("libc")]
+        private static extern long sysconf(int name);
```

Comment:
We're going to need to rationalize these test P/Invokes with the work @nguerrera is doing to P/Invoke into shims rather than directly into the underlying native library.  We can go with this for now, but we'll need to deal with this soon, as such declarations end up suffering from all of the same problems we're facing in our product code, e.g. the size of data types (the result here will probably be int on a 32-bit system, we're having to hardcode the different name values on each Unix platform, etc.)